src/ppx/grfx/vk/vk_command.cpp

```diff
@@ -1035,6 +1035,67 @@ void CommandBuffer::CopyImageToImage(
         &region);
 }
 
+void CommandBuffer::BlitImage(
+    const grfx::ImageBlitInfo* pCopyInfo,
+    grfx::Image*               pSrcImage,
+    grfx::Image*               pDstImage)
+{
+    bool isSourceDepthStencil = grfx::GetFormatDescription(pSrcImage->GetFormat())->aspect & grfx::FORMAT_ASPECT_DEPTH_STENCIL;
+    bool isDestDepthStencil   = grfx::GetFormatDescription(pDstImage->GetFormat())->aspect & grfx::FORMAT_ASPECT_DEPTH_STENCIL;
+    if (isSourceDepthStencil || isDestDepthStencil) {
+        PPX_ASSERT_MSG(pSrcImage->GetFormat() == pDstImage->GetFormat(), "both images in an image copy must be the same format if one is depth-stencil");
```

Comment:
"both images in an image copy must have the same format if either has depth or stencil"